bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+import org.openhab.binding.netatmo.internal.providers.NetatmoDescriptionProvider;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link DeviceHandler} is the abstract class that handles
+ * common behaviors of all netatmo bridges (devices)
+ *
+ * @author Gaël L'hopital - Initial contribution
+ * @author Rob Nielsen - Added day, week, and month measurements to the weather station and modules
+ *
+ */
+@NonNullByDefault
+public class DeviceHandler extends BaseBridgeHandler implements ConnectionListener {
+
+    private final Logger logger = LoggerFactory.getLogger(DeviceHandler.class);
+    private final List<AbstractChannelHelper> channelHelpers;
+
+    protected final Map<String, DeviceHandler> dataListeners = new ConcurrentHashMap<>();
+    protected final NetatmoDescriptionProvider descriptionProvider;
+    protected final ApiBridge apiBridge;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable RefreshStrategy refreshStrategy;
+
+    protected @NonNullByDefault({}) NetatmoThingConfiguration config;
+
+    public DeviceHandler(Bridge bridge, List<AbstractChannelHelper> channelHelpers, ApiBridge apiBridge,
+            NetatmoDescriptionProvider descriptionProvider) {
+        super(bridge);
+        this.apiBridge = apiBridge;
+        this.descriptionProvider = descriptionProvider;
+        this.channelHelpers = channelHelpers;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializing handler for thing {}", getThing().getUID());
+
+        config = getThing().getConfiguration().as(NetatmoThingConfiguration.class);
+
+        ModuleType supportedThingType = ModuleType.valueOf(getThing().getThingTypeUID().getId());
+        refreshStrategy = supportedThingType.refreshPeriod == RefreshPolicy.AUTO ? new RefreshStrategy(-1)
+                : supportedThingType.refreshPeriod == RefreshPolicy.CONFIG ? new RefreshStrategy(config.refreshInterval)
+                        : null;
+
+        getBridgeHandler().ifPresentOrElse(handler -> handler.registerDataListener(config.id, this),
+                () -> apiBridge.addConnectionListener(this));
+    }
+
+    @Override
+    public void apiConnectionStatusChanged(ConnectionStatus connectionStatus) {
+        if (connectionStatus.equals(ConnectionStatus.SUCCESS)) {
```

Comment:
Compare constants from left for safety:
`if (ConnectionStatus.SUCCESS.equals(connectionStatus)) {`

Reply:
Corrected.